subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java

```diff
@@ -195,25 +207,44 @@ public void jvmArgs(Iterable<?> arguments) {
         boolean xdebugFound = false;
         boolean xrunjdwpFound = false;
         boolean xagentlibJdwpFound = false;
+        JavaDebugOptions options = null;
+
         Set<Object> matches = new HashSet<Object>();
         for (Object extraJvmArg : extraJvmArgs) {
-            if (extraJvmArg.toString().equals("-Xdebug")) {
+            String extraJvmArgString = extraJvmArg.toString();
+            if (extraJvmArgString.equals("-Xdebug")) {
```

Comment:
Should we try extracting the arguments (like you've done here) or should we just emit a deprecation warning and ignore `JavaDebugOptions` when we see explicit debug options from the user?

Reply:
I focused on keeping the existing functionality and tried to make the code too smart. I think showing a warning and ignoring `DebugOptions` when the user specifies custom debug arguments is a better idea.

Reply:
I replaced the argument extracting with printing a simple warning message. I feel like the deprecation warning is inappropriate here, but I leave it to you to decide.